maroon/src/linearizer.rs

```diff
@@ -0,0 +1,101 @@
+use crate::epoch::Epoch;
+use common::range_key::UniqueU64BlobId;
+use log::debug;
+
+pub trait Linearizer {
+  fn new_epoch(&mut self, epoch: Epoch);
+}
+
+pub struct LogLineriazer {
+  sequence: Vec<UniqueU64BlobId>,
+}
+
+impl LogLineriazer {
+  pub fn new() -> LogLineriazer {
+    LogLineriazer { sequence: vec![] }
+  }
+}
+
+impl Linearizer for LogLineriazer {
+  fn new_epoch(&mut self, mut epoch: Epoch) {
+    epoch.increments.sort();
+    let new_elements_count: usize = epoch.increments.iter().map(|i| i.ids_count()).sum();
+    self.sequence.reserve(new_elements_count);
+
+    for interval in &epoch.increments {
+      for i in interval.iter() {
+        self.sequence.push(i);
+      }
+    }
+
+    debug!("new log: {:?}", self.sequence);
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use common::range_key::{
+    KeyOffset, KeyRange, U64BlobIdClosedInterval, unique_blob_id_from_range_and_offset,
+  };
+
+  #[test]
+  fn test_linear() {
+    let mut linearizer = LogLineriazer::new();
+
+    linearizer.new_epoch(Epoch::new(
```

Comment:
Why not make the `linearizer` an `Iterator` itself, so that it has its own `.next()`? For the test `.collect().into()` a `Vec<>` would do the job just fine!

Reply:
I'm not sure yet how this one will be used tbh. That's why let's not add many lines around, for now